pkg/versionstream/versionstreamrepo/gitrepo_integration_test.go

```diff
@@ -0,0 +1,70 @@
+// +build integration
+
+package versionstreamrepo_test
+
+import (
+	"testing"
+
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	RepoURL    = "https://github.com/jenkins-x/jenkins-x-versions"
+	VersionRef = "v1.0.90"
+	BranchRef  = "master"
+	HEAD       = "HEAD"
+)
+
+func TestCloneJXVersionsRepoWithATag(t *testing.T) {
+	dir, versionRef, err := versionstreamrepo.CloneJXVersionsRepo(
+		RepoURL,
+		VersionRef,
+		nil,
+		gits.NewGitCLI(),
+		true,
+		false,
+		nil,
+		nil,
+		nil,
+	)
+	assert.NoError(t, err)
+	assert.NotNil(t, dir)
+	assert.NotNil(t, versionRef)
+	assert.Equal(t, VersionRef, versionRef)
```

Comment:
For these tests we need to validate that the correct files were checked out into the directory as well

Reply:
I'm not sure I know what that means. You mean validate that we're checked out to the correct ref?

Reply:
Exactly.